service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java

```diff
@@ -320,6 +320,9 @@ public Object run() throws HiveSQLException {
             setOperationException(e);
             LOG.error("Error running hive query: ", e);
           } finally {
+            // Call Hive.closeCurrent() that closes the HMS connection, causes
+            // HMS connection leaks otherwise.
+            Hive.closeCurrent();
```

Comment:
Isn't there one place where this needs to happen?

Reply:
only here is sufficient ..other place is not required